### ISSUE TITLE
Performance profile broken

### ISSUE THREAD

**Issue by matt-kwong** [grpc/grpc#13628]
CC @ctiller 

From https://grpc-testing.appspot.com/view/Performance/job/gRPC_performance_profile_master/8957/console

```
00:10:22.527 BigQuery error in load operation: Error processing job
00:10:22.527 'grpc-testing:bqjob_r31be4e626e7c318a_0000016022c76404_1': Error while reading
00:10:22.527 data, error message: CSV table encountered too many errors, giving up. Rows: 1;
00:10:22.527 errors: 1.
00:10:22.527 Failure details:
00:10:22.527 - file-00000000: Error while reading data, error message: CSV table
00:10:22.527 references column position 1, but line starting at position:0
00:10:22.527 contains only 1 columns.
00:10:22.576 Traceback (most recent call last):
00:10:22.576   File "tools/run_tests/run_microbenchmark.py", line 215, in <module>
00:10:22.577     collectors[collect](bm_name, args)
00:10:22.577   File "tools/run_tests/run_microbenchmark.py", line 181, in collect_summary
00:10:22.577     subprocess.check_call(['bq', 'load', 'microbenchmarks.microbenchmarks', '%s.csv' % bm_name])
00:10:22.577   File "/usr/lib/python2.7/subprocess.py", line 541, in check_call
00:10:22.577     raise CalledProcessError(retcode, cmd)
00:10:22.577 subprocess.CalledProcessError: Command '['bq', 'load', 'microbenchmarks.microbenchmarks', 'bm_fullstack_unary_ping_pong.csv']' returned non-zero exit status 1
00:10:22.589 Build step 'Execute shell' marked build as failure
```

The performance profile job has been broken for almost 2 months. It looks like one of the microbenchmarks schemas might have been changed.

**Comment by ncteisen** [comment]
#12808 and #12980 both went in right when this started failing. Does any manual action need to be taken on the database when this occurs?

**Comment by matt-kwong** [comment]
Yes, https://bigquery.cloud.google.com/table/grpc-testing:microbenchmarks.microbenchmarks needs to have its fields updated. We use https://github.com/grpc/grpc/blob/master/tools/run_tests/performance/patch_scenario_results_schema.py to update the multi-language performance benchmarks; maybe we can change the script to also update the microbenchmark schemas if they change frequently enough.

**Comment by ncteisen** [comment]
I think that running 

`tools/run_tests/performance/patch_scenario_results_schema.py --bq_result_table microbenchmarks.microbenchmarks`

Should work then. I am trying it, but running into:
'oauth2client.client.HttpAccessTokenRefreshError: invalid_grant: Bad Request'

**Comment by matt-kwong** [comment]
As long as it's pointing to the right schema file, it should work. You probably have to run `gcloud auth login` to login first.

**Comment by ncteisen** [comment]
A sanity check or kokoro autocommenting warning or OWNERS file on scenario_results_json_schema.json would be good to catch these in the future. The schema will change whenever someone adds a new stat

**Comment by matt-kwong** [comment]
Good idea. I'm more of a fan of the OWNERS file, since it's usually the same group of people changing that file and approving those changes.

**Comment by jtattermusch** [comment]
@ncteisen 

the `tools/run_tests/performance/patch_scenario_results_schema.py` is only good for the results from full benchmarks, microbenchmarks use a different schema.

The way the results are uploaded to microbenchmark.microbenchmark  bq tale is pretty cryptic,
but it involves taking the json data generated by the microbenchmark and  turning it into .csv
using command that looks like this:
`tools/profiling/microbenchmarks/bm2bq.py  XYZ.counters.json XYZ.opt.json    >  bq.csv`

Then, `bq load microbenchmarks.microbenchmarks cq.csv` is used to upload the data, which fails with
```
BigQuery error in load operation: Error processing job
'grpc-testing:bqjob_r45672a7ed9fb941c_000001628f920bd2_1': Error while reading
data, error message: CSV table encountered too many errors, giving up. Rows: 1;
errors: 1. Please look into the error stream for more details.
Failure details:
- file-00000000: Error while reading data, error message: CSV table
references column position 1, but line starting at position:0
contains only 1 columns.
```

I generated an example of the data to be uploaded locally, and the format seems rather odd (not really like a CSV).
None of this seems to be documented anywhere, so I'd recommend adding some comments after we figure out what's wrong.

A snippet from the ".csv" file
```
BM_UnaryPingPong
([u'0', u'0'], ['request_size', 'response_size'])
([u'TCP', u'NoOpMutator', u'NoOpMutator'], ['fixture', 'client_mutator', 'server_mutator'])
,,2018-04-04 10:33:46,True,12,3800,release,BM_UnaryPingPong,TCP,NoOpMutator,NoOpMutator,0,0,,12491,ns,51596,51401,,14.5268,15.969,,,,,,,67.1738,385.097,,,,5.00462,1.0,1.0,2.00031,0.0,2.0,2.00063,4.00016,0.0,0.0,5.0,0.0,2.0,2.0,2.0,2.0,2.0,2.0,0.000156691,0.000548417,2.00063,0.0,0.0,0.0,13.9996,14.0047,2.00125,7.83453e-05,0.000235036,0.0,0.0,0.000235036,0.000235036,0.0,7.83453e-05,7.83453e-05,0.000156691,7.83453e-05,0.0,0.999922,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.000156691,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.000548417,0.0,2.0,0.0,0.000156691,0.0,1.99984,0.0,12.999,0.0,0.00109683,0.999922,0.0,0.0,0.0,1.00212,0.0,0.0,0.0,12.999,0.0,0.00109683,0.999922,0.0,0.0,0.0,1.00212,0.0,0.0,0.0,1.00024,0.0
```

**Comment by jtattermusch** [comment]
Fixed by #15025

**Comment by matt-kwong** [comment]
Thanks for the fix!
